snap/snapcraft.yaml

```
@@ -0,0 +1,56 @@
+name: charmed-etcd # you probably want to 'snapcraft register <name>'
+base: core24 # the base snap is the execution environment for this snap
+version: '3.5.16' # just for humans, typically '1.2+git' or '1.3.2'
+summary: etcd is a distributed reliable key-value store for distributed systems. # 79 char long summary
+description: |
+  etcd is a distributed reliable key-value store for the most critical data of a distributed system.
+  etcd is written in Go and uses the Raft consensus algorithm to manage a highly-available replicated log.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: strict # use 'strict' once you have the right plugs and slots
+
+environment:
+  SNAP_CURRENT: /snap/charmed-etcd/current
+  SNAP_DATA_CURRENT: /var/snap/charmed-etcd/current
+
+apps:
+  daemon:
+    daemon: simple
+    command: bin/etcd
+    plugs:
+      - network-bind
+    environment:
+      ETCD_CONFIG_FILE: $SNAP_CURRENT/etcd.conf.yml
```

We should refer to this file from `$SNAP_DATA_CURRENT` -- as `$SNAP_CURRENT` is a read only directory.
```suggestion
      ETCD_CONFIG_FILE: $SNAP_DATA_CURRENT/etcd.conf.yml
```